compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/node/RootNodeOwner.skiko.kt

```diff
@@ -219,8 +220,14 @@ internal class RootNodeOwner(
         onLightingInfoChanged()
     }
 
+    private fun positionOnScreen() = platformContext.convertLocalToScreenPosition(Offset.Zero) 
+
     fun invalidatePositionOnScreen() {
-        measureAndLayoutDelegate.dispatchOnPositionedCallbacks(forceDispatch = true)
+        val positionOnScreen = positionOnScreen()
+        if (positionOnScreen != currentPositionOnScreen) {
```

Comment:
Sorry but this logic is not correct for the general case. It assumes that `convertLocalToScreenPosition` might apply only translate, but it's a general API that can be used to apply scale and rotate as well

Reply:
> it's only called when the window moves.

This particular code is not for desktop only: it's exposed as a "semi" public scene API for all platforms

> That's what I was thinking too, but shouldn't just two corners be enough (top-left and bottom-right)? Can the transformation also skew the coordinate system?

it not only about skew but also rotation around the diagonal axis. But skew is just one more case.

These functions are replacement of exposing transformation matrix because for some platforms it's implementation detail, but the assumption is that they apply some matrix transformation to given coordinates.